Cleaned npm packages; Refactored Organization

aminduca opened the pull request.

aminduca committed:
*Added "repository" to package.json
https://coveord.atlassian.net/browse/DYN-898
https://coveord.atlassian.net/browse/DYN-899
https://coveord.atlassian.net/browse/DYN-900

aminduca committed:
*Removed unused (apparently boilerplate) dependencies to (1) reduce the surface of vulnerabilities, (2) have more control over transitive dependencies and to (3) clean our package.json leaving only the dependencies that are actually used.
https://coveord.atlassian.net/browse/DYN-898
https://coveord.atlassian.net/browse/DYN-899
https://coveord.atlassian.net/browse/DYN-900

lbergeron commented on package.json:
```diff
@@ -10,14 +10,19 @@
   "main": "bin/js/coveo-xrm.min.js",
   "types": "bin/js/src/Index.d.ts",
   "scripts": {
+    "setup": "npm install && npm run build",
```
I don't think the `setup` script should run the build. In fact, I would do it the other way around. In my opinion, it would make more sense to have the `build` script do `npm run setup` first. What do you think?

aminduca committed:
*Organization refactoring
https://coveord.atlassian.net/browse/DYN-882

aminduca committed:
*Organization refactoring
https://coveord.atlassian.net/browse/DYN-882

aminduca committed:
Merge branch 'master' of https://github.com/aminduca/coveo-xrm

lbergeron commented on src/context/Organization.ts:
```diff
@@ -1,45 +1,66 @@
-        return this.clientUrl && this.name
-            ? this.clientUrl.replace(this.name, discoveryUrlPrefix)
-            : this.clientUrl;
+    private getClientUrlHostnameSubpath(): string {
```
This method could be renamed to `getRegionHostName`. I think it would fit better with [Microsoft terminology](https://docs.microsoft.com/en-us/dynamics365/customer-engagement/developer/developer-resources-page).

You may also use a regular expression to capture the part of the host name that represents the region.

`/\.(crm\d*\.dynamics\.com)$/`

aminduca committed:
*getRegionHostName (rename)
https://coveord.atlassian.net/browse/DYN-898
https://coveord.atlassian.net/browse/DYN-899
https://coveord.atlassian.net/browse/DYN-900
